dashboard/app/controllers/sections_controller.rb

```diff
@@ -7,14 +7,17 @@ def show
     @secret_pictures = SecretPicture.all.shuffle
   end
 
-  # Allows you to update a section's course_id. Clears any assigned script_id
-  # in the process
+  # Allows you to update a section. Clears any assigned script_id in the process
   def update
     section = Section.find(params[:id])
     authorize! :manage, section
 
     course_id = params[:course_id]
-    script_id = params[:script_id]
+
+    # This endpoint needs to satisfy two endpoint formats for getting script_id
+    # This should be updated soon to always expect params[:script_id]
+    script_id = params[:script][:id] if params[:script]
+    script_id ||= params[:script_id]
```

Comment:
*new* - fixes bug where one consumer provides `params[:script_id]` and the other provides `params[:script][:id]`